lib/src/type_helper.dart

```diff
@@ -54,68 +60,44 @@ abstract class TypeHelper {
   /// String deserialize(DartType targetType, String expression) =>
   ///   "new ${targetType.name}.fromInt($expression)";
   /// ```.
-  String deserialize(DartType targetType, String expression);
+  // TODO(kevmoo) – document `deserializeNested`
+  String deserialize(DartType targetType, String expression,
+      String deserializeNested(DartType t, String e));
 }
 
-class JsonHelper extends TypeHelper {
-  const JsonHelper();
-
-  //TODO(kevmoo): This should be checking for toJson method, but toJson might be
-  //   gone during generation, so we'll have to check for the annotation, too!
-  // In the mean time, just assume the `canSerialize` logic will work most of
-  //   the time.
-  @override
-  bool canSerialize(DartType type) => canDeserialize(type);
+/// A [TypeChecker] for [String].
```

Comment:
[nit] Is the comment adding new information?

BTW a pattern I've seen Matan use with `TypeChecker` is to name the variable like `$String` which makes the usage look nicer than repeating `TypeChecker`